### PR TITLE
Refactor: merge docker setup file into one dockerfile & one docker-compose.yml

### PR DESCRIPTION
**Pull request by HyunggyuJang**
- Merge .dockerignore into toplevel one
- Merge Dockerfiles into toplevel one
- Simplify docker-compose.yml file
- Remove nginx setting (no more used)
  I'd like to discuss about this point. Are you okay with removing nginx setting? I understand that you want to detach client side from api server, but I found it rather cumbersome to setup in my local server env. What do you think? @danny-avila

### CODE REVIEW

**Comment by danny-avila** [comment]
have you tested it working without nginx? i've had issues without it. I will test your branch when i get the chance

**Comment by HyunggyuJang** [comment]
> have you tested it working without nginx? i've had issues without it. I will test your branch when i get the chance

Yes, indeed I use it daily.

**Review by slippersheepig — APPROVED** [review]
Just test it, works fine

**Comment by danny-avila** [comment]
any chance we can include both options in a separate dockerfile? i can simply write it in the readme

**Comment by HyunggyuJang** [comment]
> any chance we can include both options in a separate dockerfile? i can simply write it in the readme

Added nginx build recipe for docker and docker-compose file. If the user wants to use nginx, all they need to do is uncomment the client service in the docker-compose.yml file (also need to modify the api's exposed port, commented it also)

File: docker-compose.yml [hunk 1]
```diff
@@ -1,9 +1,21 @@
 version: "3.4"
 
 services:
+  # client:
+  #   image: react-client
```
Comment (HyunggyuJang) [inline docker-compose.yml]:
```suggestion
  #   image: nginx-client
```
Missed it

**Review by danny-avila — APPROVED** [review]

**Pull request merged**